modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/datalist/Datalist.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayInput} from '@clayui/form';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const Datalist = (props) => {
+	const name = props.options?.name || 'quantity_datalist';
+	const gHash = (s) => {
+		const h = 0;
+		if (s.length == 0) {
+			return h;
+		}
+		Array.from(s.length).forEach((h, i) => {
```

Comment:
`gHash` will always return `0`, as you're decomposing into an array from the string's `length`, which is a `Number`, hence resulting in `[]`. The `.forEach` will cycle through an empty array.

Also, the inner `h` has a blocked scope and it's not really altering the outer `h = 0` (if it did, it would break the `const` assignment and you should have seen an error thrown in console). So, again, this function will always return `0`. :/

In general, for as much as I see this utility useful for other purposes, in this specific case there's a little overhead. Also considering that if no one passes in a `name` prop, the default will always be `quantity_datalist` will always be the same, and so will be its resulting hash representation. Which is definitely too much overhead and i think it kinda kills the purpose of the name hashing. What do you think? @garofalog